WordPressAuthenticator/GoogleSignIn/NewGoogleAuthenticator.swift

```diff
@@ -45,24 +25,46 @@ public class NewGoogleAuthenticator: NSObject {
         clientId: GoogleClientId,
         scheme: String,
         audience: String,
-        contextProvider: ASWebAuthenticationPresentationContextProviding,
         oautTokenGetter: GoogleOAuthTokenGetting
     ) {
         self.clientId = clientId
         self.scheme = scheme
         self.audience = audience
         self.oauthTokenGetter = oautTokenGetter
-        self.contextProvider = contextProvider
     }
 
     /// Get the user's OAuth token from their Google account. This token can be used to authenticate with the WordPress backend.
-    public func getOAuthToken() async throws -> IDToken {
+    ///
+    /// The app will present the browser to hand over authentication to Google from the given `UIViewController`.
+    public func getOAuthToken(from viewController: UIViewController) async throws -> IDToken {
+        return try await getOAuthToken(
+            from: WebAuthenticationPresentationContext(viewController: viewController)
+        )
+    }
+
+    /// Get the user's OAuth token from their Google account. This token can be used to authenticate with the WordPress backend.
+    ///
+    /// The app will present the browser to hand over authentication to Google using the given
+    /// `ASWebAuthenticationPresentationContextProviding`.
+    public func getOAuthToken(
+        from contextProvider: ASWebAuthenticationPresentationContextProviding
```

Comment:
I decided to leave both `UIViewController` and `ASWebAuthenticationPresentationContextProviding` options to cater for a SwiftUI consumer, even though that is not in the roadmap as far as I'm aware.